hack/tools/test-profiling/README.md

```diff
@@ -0,0 +1,87 @@
+# Test Profiling Tools
+
+Collect and analyze heap/CPU profiles during operator-controller tests.
+
+## Quick Start
+
+```bash
+# Start profiling
+make start-profiling/baseline
+
+# Run tests
+make test-e2e
+
+# Stop and analyze
+make stop-profiling
+
+# View report
+cat test-profiles/baseline/analysis.md
+
+# Compare runs
+./bin/test-profile compare baseline optimized
+cat test-profiles/comparisons/baseline-vs-optimized.md
+```
+
+## Commands
+
+```bash
+# Build
+make build-test-profiler
+
+# Run test with profiling
+./bin/test-profile run <name> [test-target]
+
+# Start/stop daemon
+./bin/test-profile start [name]  # Daemonizes automatically
+./bin/test-profile stop
+
+# Analyze/compare
+./bin/test-profile analyze <name>
+./bin/test-profile compare <baseline> <optimized>
+./bin/test-profile collect  # Single snapshot
+```
+
+## Configuration
+
+```bash
+# Namespaces (supports components in different namespaces)
+export TEST_PROFILE_NAMESPACE=olmv1-system                       # default
+export TEST_PROFILE_OPERATOR_CONTROLLER_NAMESPACE=olmv1          # optional override
+export TEST_PROFILE_CATALOGD_NAMESPACE=catalogd-system           # optional override
+
+export TEST_PROFILE_INTERVAL=10                # seconds
+export TEST_PROFILE_CPU_DURATION=10            # seconds
+export TEST_PROFILE_MODE=both                  # both|heap|cpu
+export TEST_PROFILE_DIR=./test-profiles        # default
+export TEST_PROFILE_TEST_TARGET=test-e2e       # make target
+```
+
+## Output
+
+```
+test-profiles/
+├── <name>/
+│   ├── operator-controller/{heap,cpu}*.pprof
+│   ├── catalogd/{heap,cpu}*.pprof
+│   ├── profiler.log
+│   └── analysis.md
+└── comparisons/<name>-vs-<name>.md
+```
+
+## Interactive Analysis
+
+```bash
+cd test-profiles/<name>/operator-controller
+go tool pprof -top heap23.pprof
+go tool pprof -base=heap0.pprof -top heap23.pprof
+go tool pprof -text heap23.pprof | grep -i openapi
+```
+
+## Requirements
+
+**Runtime:**
+- Kubernetes cluster access (via kubeconfig)
+
+**Build:**
+- go 1.24+
+- make
```

Comment:
Will be hard we keep it maintained 
when we bump do we really need this section?